packages/tracing/src/browser/browsertracing.ts

```diff
@@ -0,0 +1,241 @@
+import { EventProcessor, Integration, Severity, TransactionContext } from '@sentry/types';
+import { addInstrumentationHandler, getGlobalObject, logger, safeJoin } from '@sentry/utils';
+
+import { startIdleTransaction } from '../hubextensions';
+import { DEFAULT_IDLE_TIMEOUT, IdleTransaction } from '../idletransaction';
+import { Span } from '../span';
+import { Location as LocationType } from '../types';
+
+const global = getGlobalObject<Window>();
+
+type routingInstrumentationProcessor = (context: TransactionContext) => TransactionContext;
+
+/**
+ * Gets transaction context from a sentry-trace meta.
+ */
+const setHeaderContext: routingInstrumentationProcessor = ctx => {
+  const header = getMetaContent('sentry-trace');
+  if (header) {
+    const span = Span.fromTraceparent(header);
+    if (span) {
+      return {
+        ...ctx,
+        parentSpanId: span.parentSpanId,
+        sampled: span.sampled,
+        traceId: span.traceId,
+      };
+    }
+  }
+
+  return ctx;
+};
+
+/** Options for Browser Tracing integration */
+export interface BrowserTracingOptions {
+  /**
+   * This is only if you want to debug in prod.
+   * writeAsBreadcrumbs: Instead of having console.log statements we log messages to breadcrumbs
+   * so you can investigate whats happening in production with your users to figure why things might not appear the
+   * way you expect them to.
+   *
+   * Default: {
+   *   writeAsBreadcrumbs: false;
+   * }
+   */
+  debug: {
+    writeAsBreadcrumbs: boolean;
+  };
+
+  /**
+   * The time to wait in ms until the transaction will be finished. The transaction will use the end timestamp of
+   * the last finished span as the endtime for the transaction.
+   * Time is in ms.
+   *
+   * Default: 1000
+   */
+  idleTimeout: number;
+
+  /**
+   * Flag to enable/disable creation of `navigation` transaction on history changes.
+   *
+   * Default: true
+   */
+  startTransactionOnLocationChange: boolean;
+
+  /**
+   * Flag to enable/disable creation of `pageload` transaction on first pageload.
+   *
+   * Default: true
+   */
+  startTransactionOnPageLoad: boolean;
+
+  /**
+   * beforeNavigate is called before a pageload/navigation transaction is created and allows for users
+   * to set a custom navigation transaction name. Defaults behaviour is to return `window.location.pathname`.
+   *
+   * If undefined is returned, a pageload/navigation transaction will not be created.
+   */
+  beforeNavigate(location: LocationType): string | undefined;
+
+  /**
+   * Set to adjust transaction context before creation of transaction. Useful to set name/data/tags before
+   * a transaction is sent. This option should be used by routing libraries to set context on transactions.
+   */
+  // TODO: Should this be an option, or a static class variable and passed
+  // in and we use something like `BrowserTracing.addRoutingProcessor()`
+  routingInstrumentationProcessors: routingInstrumentationProcessor[];
```

Comment:
Again, I'm not quite sure if this is right approach, would appreciate any thoughts.

Reply:
We need to be careful here, we shouldn't have another static interface again which limits how we can extend this in the future. I think it's fine this staying an option.